addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+            'internalID': invoice.name,
+        }
+        if invoice.move_type in ['out_refund', 'in_refund']:
+            eta_invoice.update({
+                'references': [invoice.reversed_entry_id.l10n_eg_uuid] if invoice.reversed_entry_id.l10n_eg_uuid else []
+            })
+        eta_invoice.update({
+            'invoiceLines': self._l10n_eg_eta_prepare_invoice_lines_data(invoice),
+            'totalDiscountAmount': invoice._get_amount_main_currency(edi_values['total_price_discount']),
+            'totalSalesAmount': invoice._get_amount_main_currency(edi_values['total_price_subtotal_before_discount']),
+            'netAmount': invoice._get_amount_main_currency(invoice.amount_untaxed),
+            'taxTotals': [{
+                'taxType': tax['l10n_eg_eta_code'].split('_')[0].upper(),
+                'amount': abs(invoice._get_amount_main_currency(tax['tax_amount_currency'])),
+            } for tax in grouped_taxes],
+            'totalAmount': invoice._get_amount_main_currency(invoice.amount_total),
+            'extraDiscountAmount': 0.0,
+            'totalItemsDiscountAmount': 0.0,
+        })
+        return eta_invoice
+
+    def _l10n_eg_eta_prepare_invoice_lines_data(self, invoice):
+        lines = []
+        for line in invoice.invoice_line_ids.filtered(lambda x: x.display_type not in ('line_section', 'line_note')):
+            discount = (line.discount / 100.0) * line.quantity * line.price_unit
+            lines.append({
+                'description': line.name,
+                'itemType': line.product_id.barcode.startswith('EG') and 'EGS' or 'GS1',
+                'itemCode': line.product_id.barcode,
+                'unitType': line.product_uom_id.l10n_eg_unit_code_id.code,
+                'quantity': line.quantity,
+                'internalCode': line.product_id.default_code or '',
+                'salesTotal': invoice._get_amount_main_currency(line.quantity*line.price_unit),
+                'total': invoice._get_amount_main_currency(line.price_total),
+                'valueDifference': 0.0,
+                'totalTaxableFees': 0.0,
+                'netTotal': invoice._get_amount_main_currency(line.price_subtotal),
+                'itemsDiscount': 0.0,
+                'unitValue': {
+                    'currencySold': invoice.currency_id.name,
+                    'amountEGP': invoice._get_amount_main_currency(line.price_unit)
+                },
+                'discount': {  # price_discount
+                    'rate': line.discount or 0.0,
+                    'amount': invoice._get_amount_main_currency(discount) or 0.0,
+                },
+                'taxableItems': [
+                    {
+                        'taxType': tax.l10n_eg_eta_code.split('_')[0].upper(),
+                        'amount': abs(invoice._get_amount_main_currency((tax.amount / 100.0) * line.price_subtotal)) or 0,
```

Comment:
this should be in the tax details as well. We should not have to recalculate this at all.